File: rxbinding/src/main/java/com/jakewharton/rxbinding2/widget/AbsListViewScrollEvent.java
package com.jakewharton.rxbinding2.widget;

import android.support.annotation.CheckResult;
import android.support.annotation.NonNull;
import android.widget.AbsListView;
import com.google.auto.value.AutoValue;

@AutoValue
public abstract class AbsListViewScrollEvent {
  @CheckResult @NonNull
  public static AbsListViewScrollEvent create(AbsListView listView, int scrollState,
      int firstVisibleItem, int visibleItemCount, int totalItemCount) {
    return new AutoValue_AbsListViewScrollEvent(listView, scrollState, firstVisibleItem,
        visibleItemCount, totalItemCount);
  }

  AbsListViewScrollEvent() {
  }

  /** The view from which this event occurred. */
  @NonNull public abstract AbsListView view();
  public abstract int scrollState();
  public abstract int firstVisibleItem();
  public abstract int visibleItemCount();
  public abstract int totalItemCount();
}

File: rxbinding/src/main/java/com/jakewharton/rxbinding2/widget/AbsListViewScrollEvent.kt
package com.jakewharton.rxbinding2.widget

import android.widget.AbsListView

data class AbsListViewScrollEvent(
  /** The view from which this event occurred.  */
  val view: AbsListView,
  val scrollState: Int,
  val firstVisibleItem: Int,
  val visibleItemCount: Int,
  val totalItemCount: Int
)
